Sources/carton/Toolchain/ToolchainManagement.swift

```diff
@@ -282,4 +282,18 @@ extension FileSystem {
 
     return AbsolutePath(binPath)
   }
+  
+  func fetchAllSwiftVersion() throws -> [String] {
+    var versions: [String] = []
+    // sdks installed by carton
+    let cartonSdkPath = homeDirectory.appending(components: ".carton", "sdk")
+    let cartonSdks = try getDirectoryContents(cartonSdkPath)
+    versions.append(contentsOf: cartonSdks)
+    // sdks installed by swiftenv
+    let swiftenvVersionPath = homeDirectory.appending(components: ".swiftenv", "versions")
```

Comment:
**suggestion:** same for this path, it's already computed on line 122, would be great to generalize it.